File: spring-reactive-02/learn-java/src/main/java/boki/learnjava/v1_newsequence/h_create/CryptoCurrencyPriceEmitter.java
package boki.learnjava.v1_newsequence.h_create;

import boki.learnjava.common.SampleData;

public class CryptoCurrencyPriceEmitter {
    private CryptoCurrencyPriceListener listener;

    public void setListener(CryptoCurrencyPriceListener listener) {
        this.listener = listener;
    }

    public void flowInto() {
        listener.onPrice(SampleData.btcPrices);
    }

    public void complete() {
        listener.onComplete();
    }
}

File: spring-reactive-02/learn-kt/src/main/kotlin/boki/learnkt/v1_newsequence/h_create/CryptoCurrencyPriceEmitter.kt
package boki.learnkt.v1_newsequence.h_create

import boki.learnjava.common.SampleData

class CryptoCurrencyPriceEmitter {
    private var listener: CryptoCurrencyPriceListener? = null

    fun setListener(listener: CryptoCurrencyPriceListener) {
        this.listener = listener
    }

    fun flowInto() {
        listener?.onPrice(SampleData.btcPrices)
    }

    fun complete() {
        listener?.onComplete()
    }
}